Rework StackNavigation into an actual Stack

yulnr opened the pull request:
Refactors the StackNavigation system into a proper **stack** with a more idiomatic API that matches patterns from other UI frameworks.

### Changes

#### New API
- `push(cx, view_id)` - Navigate to a new view (replaces show_stack_view_by_id)
- `pop(cx)` - Go back to previous view
- `pop_to_root(cx)` - Return to root view

#### Proper Navigation Stack
- Maintains complete navigation history instead of tracking only one active view
- Fixes animation issue where transitions always appeared to come from root
- Prevents navigation cycles by removing duplicate views from stack

#### Some inspection methods
- `depth()` - Get current stack depth
- `can_pop()` - Check if back navigation is possible
- `current_view()` - Get current view ID
- `stack_view_ids()` - Get complete navigation history

Annotated `show_stack_view_by_id` with `#[deprecated]` therefore the only breaking change is `StackNavigationAction::NavigateTo` being entirely replaced with `StackNavigationAction::Push(view_id)`.

#### Updated actions
```rust
// Old actions
StackNavigationAction::NavigateTo(view_id)

// New actions  
StackNavigationAction::Push(view_id)
StackNavigationAction::Pop
StackNavigationAction::PopToRoot
```

#### Support for multiple concurrent StackNavigation's, and non-full screen
- A makepad app can now have multiple instances of StackNavigation, including nested ones.
- The navigation views (`StackNavigationView`) can now be flagged with `full_screen: false` (`true` by default), to constraint the views to the StackNavigation boundaries, and disable the sliding animations. I've came across a few limitations of the Makepad animator that prevent me from supporting the animations for both full-screen and non-full screen within the same widget (unable to parametrize or update the animator target values programmatically).
Disabling full screen can be useful when using multiple StackNavigations within the same view (e.g. side-to-side in a desktop layout).

yulnr committed:
Refactor stack navigation into a proper stack

 - Replace show_stack_view_by_id with push/pop/popToRoot methods
- Add proper navigation stack with history tracking
- Fix animation transitions between consecutive views
- Add stack inspection methods (depth, can_pop, current_view)
- Maintain backward compatibility with deprecated methods
- Update action enum to use Push/Pop/PopToRoot variants

alanpoon commented:
Would you support multiple Stack navigation Widget?
https://github.com/alanpoon/makepad/tree/stack-navigation-back-to-root-multiple/examples/stack_navigation_example/src

I have 2 stack navigation Widgets. Opening the left stacknavigation view, pushes the right stack navigation to the left. 

https://github.com/user-attachments/assets/9bcb3c26-53d6-4dfb-9bce-e485427db0f7

yulnr commented:
> Would you support multiple Stack navigation Widget? https://github.com/alanpoon/makepad/tree/stack-navigation-back-to-root-multiple/examples/stack_navigation_example/src
> 
> I have 2 stack navigation Widgets. Opening the left stacknavigation view, pushes the right stack navigation to the left.
> Screen.Recording.2025-06-20.at.11.39.17.AM.mov

Hi @alanpoon, yes this is a current limitation of stacknavigation due to how it uses actions to communicate between the navigation views and the main stack navigation widget. Currently there's no proper checking of the views belonging to the right stack navigation.
It shouldn't be hard to fix, I'll take a look tomorrow.

yulnr committed:
Add support for multiple instances of StackNavigation.

- Introduced full-screen flag necessary to disable full-screen positioning and sliding animations when needed.

yulnr commented:
Hi @alanpoon I've updated the PR adding support multiple instances of StackNavigation, and also support for non-full screen (like in the video you attached) for that use case you'll have to flag your `StackNavigationView` with `full_screen: false` in the DSL, as the default behaviour is still to take over the screen. I left more details at the end of the PR description.

Here's a gist with [working example](https://gist.github.com/joulei/960634ef74eef34a3c4b1d579f62db5d) similar to the one you shared.
